alchemy/src/aws/control/resource.ts

```
@@ -0,0 +1,328 @@
+    const serviceName = pattern.split("::")[1];
+
+    try {
+      // List all resources of the service type
+      const resources = await client.listResources(`AWS::${serviceName}::*`);
+
+      // Delete each resource and track operations
+      const deletionPromises = (
+        resources.resources as CloudControlResourceInfo[]
+      ).map(async (resource) => {
+        try {
+          console.log(
+            `Deleting resource ${resource.identifier} (${resource.typeName})`
+          );
+
+          const response = await client.deleteResource(
+            resource.typeName,
+            resource.identifier
+          );
+
+          // Wait for deletion to complete
+          const result = await response;
+
+          if (result.status === "FAILED") {
+            console.error(
+              `Failed to delete resource ${resource.identifier}: ${result.message}`
+            );
+          } else {
+            console.log(`Successfully deleted resource ${resource.identifier}`);
+          }
+        } catch (error: unknown) {
+          const cloudError = error as CloudControlError;
+          if (cloudError.code === "ResourceNotFoundException") {
+            // Resource already deleted, not an error
+            console.log(`Resource ${resource.identifier} already deleted`);
+          } else {
+            console.error(
+              `Error deleting resource ${resource.identifier}:`,
+              error
+            );
+          }
+        }
+      });
+
+      // Wait for all deletions to complete
+      await Promise.allSettled(deletionPromises);
+    } catch (error: unknown) {
+      const cloudError = error as CloudControlError;
+      if (cloudError.code === "ResourceNotFoundException") {
+        // No resources found, not an error
+        console.log(`No resources found matching pattern ${pattern}`);
+      } else {
+        console.error(`Error listing resources for pattern ${pattern}:`, error);
```

Should we throw? What errors are swallowed here ?